src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs

```diff
@@ -2,36 +2,60 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 using System;
 using System.Collections.Immutable;
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.CodeActions;
-using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
 using Microsoft.CodeAnalysis.CSharp.ConvertTupleToStruct;
 using Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles;
-using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeActions;
-using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
 using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
 using Microsoft.CodeAnalysis.NamingStyles;
 using Microsoft.CodeAnalysis.Remote.Testing;
 using Microsoft.CodeAnalysis.Simplification;
 using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Testing;
 using Roslyn.Test.Utilities;
 using Xunit;
 
 namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ConvertTupleToStruct
 {
-    public class ConvertTupleToStructTests : AbstractCSharpCodeActionTest
+    using VerifyCS = CSharpCodeRefactoringVerifier<CSharpConvertTupleToStructCodeRefactoringProvider>;
+
+    [UseExportProvider]
+    public class ConvertTupleToStructTests
     {
-        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
-            => new CSharpConvertTupleToStructCodeRefactoringProvider();
+        #region update containing member tests
 
-        protected override ImmutableArray<CodeAction> MassageActions(ImmutableArray<CodeAction> actions)
-            => FlattenActions(actions);
+        private static OptionsCollection PreferImplicitTypeWithInfo()
+            => new OptionsCollection(LanguageNames.CSharp)
+            {
+                { CSharpCodeStyleOptions.VarElsewhere, true, NotificationOption2.Suggestion },
+                { CSharpCodeStyleOptions.VarWhenTypeIsApparent, true, NotificationOption2.Suggestion },
+                { CSharpCodeStyleOptions.VarForBuiltInTypes, true, NotificationOption2.Suggestion },
+            };
 
-        #region update containing member tests
+        private static async Task Test(
+            string text,
+            string expected,
+            int index = 0,
+            OptionsCollection? options = null,
+            TestHost testHost = TestHost.InProcess,
+            string[]? actions = null)
+        {
+            var test = new VerifyCS.Test
+            {
+                TestCode = text,
+                FixedCode = expected,
+                TestHost = testHost,
+                CodeActionIndex = index,
```

Comment:
I'd use both CodeActionIndex and CodeActionEquivalenceKey

Reply:
these are refactorign.  so as per our default, we don't set equivalence keys (since we dont' have any scenarios we expect to use them).

Reply:
Marking this one as active. When testing code actions, the `CodeActionIndex` is optional but the `CodeActionEquivalenceKey` is required (and cannot be null). Generally both are used together. This approach substantially improves readability of the tests, and hardens the tests against clarity issues that can occur if the ordering changes. The implementation may need to be updated to meet this requirement.

Reply:
k, will do.